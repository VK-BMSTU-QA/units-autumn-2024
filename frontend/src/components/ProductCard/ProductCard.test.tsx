import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';
import { Product } from "../../types";
import * as getPriceUtil from '../../utils/getPrice';

describe('ProductCard test', () => {
    const product: Product = {
        id: 1,
        name: 'Product 1',
        description: 'best product',
        price: 777,
        priceSymbol: '₽',
        category: 'Для дома',
        imgUrl: '/product.png',
    };

    it('should render correctly', () => {
        const { asFragment } = render(<ProductCard {...product} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly without image', () => {
        const productWithoutImg = { ...product, imgUrl: undefined };
        const { asFragment } = render(<ProductCard {...productWithoutImg} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should not render img', () => {
        const rendered = render(
            <ProductCard
                id={1}
                name={'afanas'}
                description={'hello55'}
                price={777}
                category={'Электроника'}
            />
        );

        expect(rendered.queryByRole('img')).toBeNull();
    });

    it('should call once getPrice util', () => {
        const mockGetPrice = jest.spyOn(getPriceUtil, 'getPrice');
        expect(mockGetPrice).not.toBeCalled();
        render(
            <ProductCard
                id={1}
                name={'niceday55'}
                description={'ananas'}
                price={333}
                category={'Для дома'}
            />
        );
        expect(mockGetPrice).toBeCalledTimes(1);
    });
});
