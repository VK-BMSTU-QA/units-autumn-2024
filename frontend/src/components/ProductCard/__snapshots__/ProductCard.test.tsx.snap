// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Product 1
      </h2>
      <div
        class="product-card__description"
      >
        best product
      </div>
      <h1
        class="product-card__price"
      >
        777 ₽
      </h1>
      <div
        class="product-card__category"
      >
        Для дома
      </div>
    </div>
    <img
      alt="Product 1"
      class="product-card__image"
      src="/product.png"
    />
  </div>
</DocumentFragment>
`;

exports[`ProductCard test should render correctly without image 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        Product 1
      </h2>
      <div
        class="product-card__description"
      >
        best product
      </div>
      <h1
        class="product-card__price"
      >
        777 ₽
      </h1>
      <div
        class="product-card__category"
      >
        Для дома
      </div>
    </div>
  </div>
</DocumentFragment>
`;
